lib/Db/RecipeDb.php

```diff
@@ -310,7 +310,7 @@ public function findRecipes(array $keywords, string $user_id) {
 		$qb->setParameters($params, $types);
 		$qb->setParameter('user', $user_id, TYPE::STRING);
 
-		$qb->groupBy(['r.name', 'r.recipe_id']);
+		$qb->groupBy(['r.name', 'r.recipe_id', 'k.name']);
```

Comment:
Why are you grouping by the keyword? A single keyword should be only uniquely in the DB associated with a recipe. So I do not get the intention behind this change.

Reply:
Otherwise each recipe is only returned with a single keyword.

Compare, e.g.,

```
SELECT `r`.`recipe_id`, `r`.`name`, `k`.`name` as `keywords` 
FROM `oc_cookbook_names` `r` 
LEFT JOIN `oc_cookbook_keywords` `k` ON k.recipe_id = r.recipe_id 
LEFT JOIN `oc_cookbook_categories` `c` ON r.recipe_id = c.recipe_id 
WHERE (r.user_id = "admin") 
GROUP BY `r`.`name`, `r`.`recipe_id`, `k`.`name` ORDER BY `r`.`name` ASC;
```

with

```
SELECT `r`.`recipe_id`, `r`.`name`, `k`.`name` as `keywords` 
FROM `oc_cookbook_names` `r` 
LEFT JOIN `oc_cookbook_keywords` `k` ON k.recipe_id = r.recipe_id 
LEFT JOIN `oc_cookbook_categories` `c` ON r.recipe_id = c.recipe_id 
WHERE (r.user_id = "admin") 
GROUP BY `r`.`name`, `r`.`recipe_id` ORDER BY `r`.`name` ASC;
```

Reply:
You should have recipes with multiple keywords in your database to see the difference

Reply:
Yes, you are completely right.